contracts/hydro/src/token_manager.rs

```diff
@@ -318,22 +334,62 @@ impl TokenInfoProviderLSM {
     }
 }
 
+#[cw_serde]
+pub struct TokenInfoProviderBase {
+    pub token_group_id: String,
+    pub denom: String,
+    pub ratio: Decimal,
+}
+
+impl TokenInfoProviderBase {
+    // Returns OK if the denom is same as the denom of the base Hydro token (e.g. ATOM, OSMO, etc.).
+    pub fn resolve_denom(&mut self, denom: String) -> StdResult<String> {
```

Comment:
Let's keep the `TokenInfoProvider` enum functions interface, even if some parameters aren't used by the `TokenInfoProviderBase`. So:

`resolve_denom(&mut self, _deps: &Deps<NeutronQuery>, _round_id: u64, _denom: String)`